File: spring/src/test/java/com/example/PostRepositoryTest.java
package com.example;

import io.quarkus.test.junit.QuarkusTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

@QuarkusTest
public class PostRepositoryTest {

    @Inject
    PostRepository postRepository;

    @Test
    public void testSave() {
        Post post = Post.of("test", "test content");
        var saved = postRepository.save(post);
        assertThat(saved).isNotNull();

        var postById = postRepository.findById(saved.id);
        assertThat(postById.isPresent()).isTrue();

        var data = postById.get();
        assertThat(data.title).isEqualTo("test");
    }
}

File: mongodb-kotlin/src/test/kotlin/com/example/PostRepositoryTest.kt
package com.example

import io.quarkus.test.junit.QuarkusTest
import jakarta.inject.Inject
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.logging.Level
import java.util.logging.Logger

@QuarkusTest
class PostRepositoryTest {
    companion object {
        private val log = Logger.getLogger(PostRepositoryTest::class.java.name)
    }


    @Inject
    lateinit var postRepository: PostRepository

    @Test
    fun `test save and query`() {
        val entity = Post(title = "foo", body = "bar")
        postRepository.persist(entity)

        log.log(Level.INFO, "entity: $entity")
        assertThat(entity).isNotNull()

        val post = postRepository.findByTitle("foo")
        log.log(Level.INFO, "found post: $post")
        assertThat(post!!.title).isEqualTo("foo")

    }
}